test/integration/apiserver/tracing/tracing_test.go

```diff
@@ -354,8 +354,8 @@ endpoint: %s`, listener.Addr().String())), os.FileMode(0755)); err != nil {
 	traceservice.RegisterTraceServiceServer(srv, fakeServer)
 
 	go func() {
-		if err = srv.Serve(listener); err != nil {
-			t.Error(err)
+		if serveErr := srv.Serve(listener); serveErr != nil {
+			t.Error(serveErr)
```

Comment:
It's probably better to explicitly return in this gofunc like the one above, no?